packages/tao/src/shared/nx-plugin.ts

```diff
@@ -1,10 +1,18 @@
 import { sync } from 'fast-glob';
 import { existsSync } from 'fs';
-import { dirname, join } from 'path';
+import * as path from 'path';
+import { register } from 'ts-node';
```

Comment:
It may be worth looking at [esbuild-register](https://github.com/egoist/esbuild-register) instead of ts-node.

It allows separation of the TypeScript compiler (for Type checking) vs running typescript code. Since I have moved in this direction the DX of my repos have improved a lot because I can run typescript in watch mode while running things is super quick

Reply:
We will be swapping back to swc-node to be in line with workspace lint rules after #9142 is merged.